File: packages/typescript/test/contract.test.ts
import { readFileSync } from "node:fs";

import { describe, expect, it } from "vitest";

import { DOMAIN, VERSION } from "../src/index.js";

const vector = JSON.parse(
  readFileSync(new URL("../../../tests/contract/package.json", import.meta.url), "utf-8"),
) as { domain: string; version: string };

describe("契约向量", () => {
  it("领域名与向量一致", () => {
    expect(DOMAIN).toBe(vector.domain);
  });

  it("版本与向量一致", () => {
    expect(VERSION).toBe(vector.version);
  });
});

File: packages/dart/test/contract_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:quanttide_code/quanttide_code.dart';
import 'package:test/test.dart';

// dart test 的工作目录是包根（packages/dart），向量在工具箱根下
Map<String, dynamic> _vector() => jsonDecode(
      File('../../tests/contract/package.json').readAsStringSync(),
    ) as Map<String, dynamic>;

void main() {
  test('领域名与契约向量一致', () {
    expect(domain, _vector()['domain']);
  });

  test('版本与契约向量一致', () {
    expect(version, _vector()['version']);
  });
}
